src/lib/reflect.ts

```diff
@@ -585,6 +586,17 @@ export class ReflectedGenericRef extends ReflectedTypeRef<format.RtGenericType>
     override matchesValue(value: any, errors?: Error[], context?: string): boolean {
         return this.baseType.matchesValue(value, errors, context);
     }
+
+    matchesRef(that: ReflectedGenericRef): boolean {
+        // return deepEqual(this, that) // is this too lazy?
```

Comment:
It probably would roughly work for generics, but definitely would be incorrect if we, for instance, added the filename that the type reference came from (which shouldnt be compared). There could be other things that differ between the objects that don't need to be compared.